### PR TITLE
Fix document regarding ArtifactNotFound

### PR DESCRIPTION
**Pull request by smygw72**
## Motivation

Some old documents remains (e.g., `:exc:~optuna_dashboard.artifact.exceptions.ArtifactNotFound exception` ). 

## Description of the changes

Updated the above documents.

### CODE REVIEW

File: optuna/artifacts/_protocol.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def open_reader(self, artifact_id: str) -> BinaryIO:
 
         This method should return a binary file-like object in read mode, similar to
         ``open(..., mode="rb")``. If the artifact does not exist, an
-        :exc:`~optuna_dashboard.artifact.exceptions.ArtifactNotFound` exception
+        :exc:`~optuna.artifact.exceptions.ArtifactNotFound` exception
```
Comment (toshihikoyanase) [inline optuna/artifacts/_protocol.py:29]: A nit.
```suggestion
        :exc:`~optuna.artifacts.exceptions.ArtifactNotFound` exception
```

File: optuna/artifacts/_protocol.py [hunk 2]
```diff
@@ -50,7 +50,7 @@ def remove(self, artifact_id: str) -> None:
         """Remove the artifact identified by the artifact_id.
 
         This method should delete the artifact from the backend. If the artifact does not
-        exist, an :exc:`~optuna_dashboard.artifact.exceptions.ArtifactNotFound` exception
+        exist, an :exc:`~optuna.artifact.exceptions.ArtifactNotFound` exception
```
Comment (toshihikoyanase) [inline optuna/artifacts/_protocol.py:53]: ditto.
```suggestion
        exist, an :exc:`~optuna.artifacts.exceptions.ArtifactNotFound` exception
```

File: optuna/artifacts/exceptions.py [hunk 3]
```diff
@@ -5,8 +5,8 @@ class ArtifactNotFound(OptunaError):
     """Exception raised when an artifact is not found.
 
     It is typically raised while calling
-    :meth:`~optuna_dashboard.artifact.protocol.ArtifactBackend.open` or
-    :meth:`~optuna_dashboard.artifact.protocol.ArtifactBackend.remove` methods.
+    :meth:`~optuna.artifact.protocol.ArtifactBackend.open_reader` or
```
Comment (toshihikoyanase) [inline optuna/artifacts/exceptions.py:8]: Hmm, I'm not sure but ArtifactStore is in a private submodule.
```suggestion
    :meth:`~optuna.artifacts._protocol.ArtifactStore.open_reader` or
```

File: optuna/artifacts/exceptions.py [hunk 4]
```diff
@@ -5,8 +5,8 @@ class ArtifactNotFound(OptunaError):
...
+    :meth:`~optuna.artifact.protocol.ArtifactBackend.remove` methods.
```
Comment (toshihikoyanase) [inline optuna/artifacts/exceptions.py:9]:
```suggestion
    :meth:`~optuna.artifacts._protocol.ArtifactStore.remove` methods.
```

**Review by toshihikoyanase — CHANGES_REQUESTED** [review]
Could you check the module paths, please?

**Review by c-bata — APPROVED** [review]
LGTM

**Review by toshihikoyanase — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**